File: spikes/issues/86/js/probe.mjs
// Spike for issue #86 (JS): is a toolkit-less ask possible on this branch?
// Claim under test: "Fixed in JS on the jev branch: a synchronous Toolkit.empty()".
import { createClient, createToolkit, Toolkit } from "../../../../js/dist/index.js"

const base = process.env.SPIKE86_BASE
const client = createClient({ baseUrl: base, style: "openai", model: "mock", apiKey: "not-a-real-key" })

// 1. Does a synchronous Toolkit.empty() exist at all?
console.log("js: typeof Toolkit.empty =", typeof Toolkit?.empty)

// 2. The call the issue says should work: no toolkit.
try {
  const r = await client.ask("write me a haiku", {})
  console.log("js: no-toolkit ask OK, text=", JSON.stringify(r.text))
} catch (e) {
  console.log("js: no-toolkit ask FAILED:", e.constructor.name + ": " + e.message)
}

// 3. Not even a ctx at all (the shape a completion user reaches for first).
try {
  const r = await client.ask("write me a haiku")
  console.log("js: bare ask OK, text=", JSON.stringify(r.text))
} catch (e) {
  console.log("js: bare ask FAILED:", e.constructor.name + ": " + e.message)
}

// 4. The workaround that DOES exist today: async, but it does express "no tools".
const tk = await createToolkit({ builtins: false })
const r = await client.ask("write me a haiku", { toolkit: tk })
console.log("js: createToolkit({builtins:false}) OK, tools=", tk.tools().length, "text=", JSON.stringify(r.text))

File: spikes/issues/86/py/probe.py
"""Spike for issue #86 (Python): a toolkit-less run/ask."""
import asyncio, os, sys
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", "python", "src"))
from toolnexus import create_client, create_toolkit  # noqa: E402


async def main() -> None:
    base = os.environ["SPIKE86_BASE"]
    c = create_client(base_url=base, style="openai", model="mock", api_key="not-a-real-key")

    # 1. The call a completion user reaches for.
    try:
        r = await c.run("write me a haiku")
        print("py: no-toolkit run OK, text=", repr(r.text))
    except TypeError as e:
        print("py: no-toolkit run FAILED:", type(e).__name__ + ": " + str(e))

    # 2. Passing None explicitly — does the loop survive it?
    try:
        r = await c.run("write me a haiku", None)
        print("py: None-toolkit run OK, text=", repr(r.text))
    except Exception as e:
        print("py: None-toolkit run FAILED:", type(e).__name__ + ": " + str(e))

    # 3. The workaround that exists today.
    tk = await create_toolkit(builtins=False)
    r = await c.run("write me a haiku", tk)
    print("py: create_toolkit(builtins=False) OK, tools=", len(tk.tools()), "text=", repr(r.text))


asyncio.run(main())
